crates/storage/provider/src/traits/trie.rs

```diff
@@ -0,0 +1,23 @@
+use crate::BundleStateWithReceipts;
+use auto_impl::auto_impl;
+use reth_interfaces::provider::ProviderResult;
+use reth_primitives::B256;
+use reth_trie::updates::TrieUpdates;
+
+/// A type that can compute the state root of a given post state.
+#[auto_impl(&, Box, Arc)]
+pub trait StateRootProvider: Send + Sync {
+    /// Returns the state root of the `BundleState` on top of the current state.
+    ///
+    /// NOTE: It is recommended to provide a different implementation from
+    /// `state_root_with_updates` since it affects the memory usage during state root
+    /// computation.
```

Comment:
smol doc nit

```suggestion
    /// # Note
    ///
    /// It is recommended to provide a different implementation from
    /// `state_root_with_updates` since it affects the memory usage during state root
    /// computation.
```